### PR TITLE
Add transpose_conv, sorting and searching operator benchmarks to Opperf

### PR DESCRIPTION
**Pull request by sandeep-krishnamurthy**
## Description ##

Add Transpose Conv, sorting and searching ops for opperf.
deconvolution, sort, topk, argmax, argmin, argsort

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] Code is well-documented: 
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] Add Deconvolution, sorting and searching ops in opperf

@apeforest @ChaiBapchya

### CODE REVIEW

**Comment by ChaiBapchya** [comment]
Apart from that weird Travis build error (which I guess will go by retriggering), LGTM!

File: benchmark/opperf/nd_operations/nn_conv_operators.py [hunk 1]
```diff
@@ -135,3 +135,50 @@ def run_convolution_operators_benchmarks(ctx=mx.cpu(), dtype='float32', warmup=2
...
+                                                                        "pad": (0, 0),
+                                                                        "num_filter": 64,
+                                                                        "no_bias": False,
+                                                                        "layout": 'NCHW'}
+                                                                       ],
+                                                               warmup=warmup,
+                                                               runs=runs)
+    # Prepare combined results
+    mx_transpose_conv_op_results = merge_map_list(conv1d_transpose_benchmark_res + conv2d_transpose_benchmark_res)
+    return mx_transpose_conv_op_results
```
Comment (apeforest) [inline benchmark/opperf/nd_operations/nn_conv_operators.py]: nit: add new line to the end of file.

File: benchmark/opperf/nd_operations/nn_conv_operators.py [hunk 2]
```diff
@@ -135,3 +135,50 @@ def run_convolution_operators_benchmarks(ctx=mx.cpu(), dtype='float32', warmup=2
...
+    for conv_data in [(32, 3, 256), (32, 3, 64)]:
+        conv1d_transpose_benchmark_res += run_performance_test([getattr(MX_OP_MODULE, "Deconvolution")],
+                                                               run_backward=True,
+                                                               dtype=dtype,
+                                                               ctx=ctx,
+                                                               inputs=[{"data": conv_data,
+                                                                        "weight": (3, 64, 3),
+                                                                        "bias": (64,),
+                                                                        "kernel": (3,),
+                                                                        "stride": (1,),
```
Comment (apeforest) [inline benchmark/opperf/nd_operations/nn_conv_operators.py]: These parameters such as stride, dilate and pad seem to make impact to performance. Would it make sense to test those variations?

File: benchmark/opperf/utils/op_registry_utils.py [hunk 3]
```diff
@@ -240,6 +242,26 @@ def get_all_reduction_operators():
...
 
 
+def get_all_sorting_searching_operators():
+    """Gets all Sorting and Searching operators registered with MXNet.
+
+    Returns
+    -------
+    {"operator_name": {"has_backward", "nd_op_handle", "params"}}
+    """
+    sort_search_ops = ['sort', 'argsort', 'argmax', 'argmin', 'topk']
```
Comment (apeforest) [inline benchmark/opperf/utils/op_registry_utils.py]: If `argmax` and `argmin` are treated as sort search ops, why not `max` and `min`?

Comment (ChaiBapchya) [reply benchmark/opperf/utils/op_registry_utils.py]: `max` and `min` have already been covered in reduction op
In that sense, `argmax` and `argmin` can be placed in 
https://github.com/apache/incubator-mxnet/blob/b88705e7744be1cc22122eb2c5b492e495ab45bb/benchmark/opperf/nd_operations/reduction_operators.py#L24

Comment (ChaiBapchya) [reply benchmark/opperf/utils/op_registry_utils.py]: NDArray API shows - argmax and argmin in the "Array sorting and searching" section.
So I guess this distinction is correct
Sort & Search - `argmax`, `argmin`
Reduction - `max`, `min`
Screen Shot 2019-07-15 at 4 13 36 PM
https://mxnet.incubator.apache.org/api/python/ndarray/ndarray.html#array-sorting-and-searching

Comment (sandeep-krishnamurthy) [reply benchmark/opperf/utils/op_registry_utils.py]: Lin, reasoning was same as described above the Chai.

File: benchmark/opperf/nd_operations/nn_conv_operators.py (hunk 2 above)
Comment (sandeep-krishnamurthy) [reply benchmark/opperf/nd_operations/nn_conv_operators.py]: Agreed, for now, I used configuration used in standard architecture to get started. I was planning to cover more variations in next phase with possible automation around input generation. Is that okay?

File: benchmark/opperf/nd_operations/nn_conv_operators.py (hunk 1 above)
Comment (sandeep-krishnamurthy) [reply benchmark/opperf/nd_operations/nn_conv_operators.py]: Done.

**Pull request merged**